lib/shopify_app/login_protection.rb

```diff
@@ -1,44 +1,46 @@
-module ShopifyApp::LoginProtection
-  extend ActiveSupport::Concern
+module ShopifyApp
+  module LoginProtection
+    extend ActiveSupport::Concern
 
-  included do
-    rescue_from ActiveResource::UnauthorizedAccess, :with => :close_session
-  end
+    included do
+      rescue_from ActiveResource::UnauthorizedAccess, :with => :close_session
+    end
 
-  def shopify_session
-    if shop_session
-      begin
-        # session[:shopify] set in LoginController#show
-        ShopifyAPI::Base.activate_session(shop_session)
-        yield
-      ensure
-        ShopifyAPI::Base.clear_session
+    def shopify_session
+      if shop_session
+        begin
+          ShopifyAPI::Base.activate_session(shop_session)
+          yield
+        ensure
+          ShopifyAPI::Base.clear_session
+        end
```

Comment:
this is a good point - this is how it runs now but yeah it really should use `ShopifyAPI::Session.temp`